### ISSUE TITLE
Save core_root after helix submission for coreclr jobs.

### ISSUE THREAD

**Issue by jashook** [dotnet/runtime#36398]
Currently getting the failing product bits requires downloading the correlation payload, which most people do not know how to do.

**Comment by jkoritzinsky** [comment]
@MattGal there used to be a tool called HelixReproCore that we could use to pull down correlation and work item payloads to repro tests. Last time I used it it seemed to incorrectly extract a multi-folder correlation payload. Is there an alternative to this tool that's more up to date?

**Comment by MattGal** [comment]
> @MattGal there used to be a tool called HelixReproCore that we could use to pull down correlation and work item payloads to repro tests. Last time I used it it seemed to incorrectly extract a multi-folder correlation payload. Is there an alternative to this tool that's more up to date?

That'd be `\\helixutil\tools\Repro\HelixJobReproCore.dll`

The tool is unsupported, though with managerial support I could easily port it over (indeed, it was written before the Arcade Helix SDK, which would greatly simplify it!).

As for it not working with destination payloads, how long ago are we talking?  I believe I added support for correlation payloads with destinations almost a year ago. If you can share a correlation id of a job that doesn't work with the above app I'll see about fixing it up for you.

**Comment by jkoritzinsky** [comment]
It was job 67eea375-f90c-482c-8667-0fc268877691 and work item `Microsoft.NET.Build.Tasks.dll.2` that I noticed it. The subfolders of the correlation payload were extracted, but they were extracted to the root folder instead of to their requisite subfolders. This was about 4.5 months ago.

**Comment by MattGal** [comment]
You're right, the version as on that share had a bug where it was unzipping the named payloads one folder up from where it should ... alas that version was never funded nor checked in and the place where its source lives was on a Surface Book whose battery exploded and got replaced.  

I looked into resurrecting it as a 3.1 app but I don't have the right sources for it so I may make that as a project for another day (I'm thinking a tool that really just fetches payloads, as that seems to be the primary usage.

Please do feel encouraged to file feature requests in dotnet/core-eng for a console repro tool.  My prototype became the repro tool we had for a while that made VMs, but that had some security concerns and was removed.
